app/src/main/java/tech/pauly/findapet/discover/DiscoverErrorViewModel.kt

```diff
@@ -0,0 +1,76 @@
+package tech.pauly.findapet.discover
+
+import android.arch.lifecycle.Lifecycle
+import android.arch.lifecycle.OnLifecycleEvent
+import android.databinding.ObservableBoolean
+import android.databinding.ObservableInt
+import tech.pauly.findapet.R
+import tech.pauly.findapet.data.models.StatusCode
+import tech.pauly.findapet.shared.BaseViewModel
+import tech.pauly.findapet.shared.datastore.TransientDataStore
+import tech.pauly.findapet.shared.datastore.UseCase
+import javax.inject.Inject
+
+open class DiscoverErrorViewModel @Inject
+constructor(private val dataStore: TransientDataStore) : BaseViewModel() {
+
+    var errorVisible = ObservableBoolean(false)
+    var errorTitle = ObservableInt(R.string.missing)
+    var errorBody1 = ObservableInt(R.string.missing)
+    var errorBody2 = ObservableInt(R.string.missing)
+
+    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
+    fun subscribeToErrorEvents() {
+        dataStore.observeAndGetUseCase(DiscoverErrorUseCase::class)
+                .subscribe({
+                    if (it.statusCode == null) hideError() else showError(it.statusCode)
+                }, Throwable::printStackTrace)
+                .onLifecycle()
+    }
+
+    private fun showError(statusCode: StatusCode) {
+        errorTitle.set(when(statusCode) {
+            StatusCode.ERR_NO_ANIMALS -> R.string.no_animals_title
```

Comment:
Would just put this stuff in the enum itself if they were all unique and there wasn't one generic fallback. This is testable anyway so /shrug